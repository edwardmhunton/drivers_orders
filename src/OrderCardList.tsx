import React from "react";

import { OrderCard } from "./OrderCard.tsx";
import { List, ListItem } from "@material-ui/core";

const styles = {
  display: "flex",
  flexDirection: "column",
  alignItems: "flex-start"
};

export const OrderCardList = ({
  orders,
  onPress,
  changeOrderLocation,
  currentSelectedOrder
}) => {
  return (
    <List style={styles} className="ordercardlist">
      {orders.map((item, index) => {
        return (
          <ListItem key={index}>
            <OrderCard
              k={index}
              order={item}
              onPress={onPress}
              selected={
                !currentSelectedOrder || currentSelectedOrder.id !== item.id
                  ? false
                  : true
              }
              changeOrderLocation={changeOrderLocation}
            />
          </ListItem>
        );
      })}
    </List>
  );
};
