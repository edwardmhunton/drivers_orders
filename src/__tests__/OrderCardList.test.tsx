import React from "react";

import renderer from "react-test-renderer";

import { OrderCardList } from "../OrderCardList";

const ordersData = {
  orders: [
    {
      id: "12345",
      PickupTime: "1556548991000",
      PickupAddress: {
        coordinates: { lat: "51.5325764", lng: "-0.0578597" },
        street_address: {
          post_code: "EX203LZ",
          address_line_1: "76 Kenworthy Road",
          city: "London"
        }
      },

      PackageSize: {
        length: "1000",
        width: "1000",
        height: "1000"
      }
    },
    {
      id: "678910",
      PickupTime: "1556548991000",
      PickupAddress: {
        coordinates: { lat: "51.55423", lng: "-0.03238" },
        street_address: {
          post_code: "EX203LZ",
          address_line_1: "76 Kenworthy Road",
          city: "London"
        }
      },
      PackageSize: {
        length: "1000",
        width: "1000",
        height: "1000"
      }
    }
  ]
};

describe("the fist test", () => {
  it("Renders Correctly", () => {
    const OrderCardListComponent = renderer
      .create(<OrderCardList orders={ordersData.orders} />)
      .toJSON();
    expect(OrderCardListComponent).toMatchSnapshot();
  });
});
