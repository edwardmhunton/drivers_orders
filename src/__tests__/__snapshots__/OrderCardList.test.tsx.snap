// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderCardList test Renders Correctly 1`] = `
<div
  className="ordercardlist"
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
    }
  }
>
  <div
    className="card"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "white",
        "color": "grey",
        "display": "flex",
        "flexDirection": "column",
        "fontSize": 9,
        "height": 200,
        "width": 200,
      }
    }
  >
    <div
      className="card-body"
    >
      <h5
        className="card-title"
      >
        Order 12345
         
      </h5>
      <p>
        Pickup Time: Mon Apr 29 2019
      </p>
      <div>
        <p>
          Pickup Address (free text)
        </p>
        <form>
          <input
            defaultValue="51.5325764, -0.0578597"
            onBlur={[Function]}
            type="text"
          />
        </form>
      </div>
      <div>
        <div>
          <div>
            Package Dimensions
          </div>
          <div>
            W: 1000mm
          </div>
          <div>
            H: 1000mm
          </div>
          <div>
            L: 1000mm
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="card"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "white",
        "color": "grey",
        "display": "flex",
        "flexDirection": "column",
        "fontSize": 9,
        "height": 200,
        "width": 200,
      }
    }
  >
    <div
      className="card-body"
    >
      <h5
        className="card-title"
      >
        Order 678910
         
      </h5>
      <p>
        Pickup Time: Mon Apr 29 2019
      </p>
      <div>
        <p>
          Pickup Address (free text)
        </p>
        <form>
          <input
            defaultValue="51.55423, -0.03238"
            onBlur={[Function]}
            type="text"
          />
        </form>
      </div>
      <div>
        <div>
          <div>
            Package Dimensions
          </div>
          <div>
            W: 1000mm
          </div>
          <div>
            H: 1000mm
          </div>
          <div>
            L: 1000mm
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
